pkg/operator/ceph/cluster/mon/mon.go

```diff
@@ -1791,3 +1801,72 @@ func (c *Cluster) releaseOrchestrationLock() {
 	c.orchestrationMutex.Unlock()
 	logger.Debugf("Released lock for mon orchestration")
 }
+
+func (c *Cluster) RotateMonCephxKeys() (bool, error) {
+	clusterObj := &cephv1.CephCluster{}
+	if err := c.context.Client.Get(c.ClusterInfo.Context, c.ClusterInfo.NamespacedName(), clusterObj); err != nil {
+		return false, errors.Wrapf(err, "failed to get cluster %v.", c.ClusterInfo.NamespacedName())
+	}
+
+	desiredCephVersion := c.ClusterInfo.CephVersion
+	// TODO: for rotation WithCephVersionUpdate fix this to have the right runningCephVersion and desiredCephVersion
+	runningCephVersion := c.ClusterInfo.CephVersion
+
+	shouldRotateMonKeys, err := keyring.ShouldRotateCephxKeys(clusterObj.Spec.Security.CephX.Daemon, runningCephVersion, desiredCephVersion, *clusterObj.Status.Cephx.Mon)
+	if err != nil {
+		return false, errors.Wrapf(err, "failed to check if mon daemon keys should be rotated in the namespace %q", c.ClusterInfo.Namespace)
+	}
+
+	if !shouldRotateMonKeys {
+		logger.Debugf("cephx key rotation for mon daemon in the namespace %q is not required", c.ClusterInfo.Namespace)
+		return shouldRotateMonKeys, nil
+	}
+
+	logger.Infof("cephx keys for mon daemons in the namespace %q will be rotated", c.ClusterInfo.Namespace)
+
+	k := keyring.GetSecretStore(c.context, c.ClusterInfo, c.ClusterInfo.OwnerInfo)
+	newKey, err := k.RotateKey(controller.MonCephxUser)
```

Comment:
if the reconcile is failed to update the rotation, it will re-rotate,
I think its true for other daemons too.

Reply:
cc @BlaineEXE ^^